builder/proxmox/bootcommand_driver.go

```diff
@@ -78,37 +89,74 @@ func NewProxmoxDriver(c commandTyper, vmRef *proxmox.VmRef, interval time.Durati
 }
 
 func (p *proxmoxDriver) SendKey(key rune, action bootcommand.KeyAction) error {
-	if special, ok := p.runeMap[key]; ok {
-		return p.send(special)
+	switch action.String() {
+	case "Press":
+		if special, ok := p.runeMap[key]; ok {
+			return p.send(special)
+		}
+		var keys string
+		if unicode.IsUpper(key) {
+			keys = fmt.Sprintf("shift-%c", unicode.ToLower(key))
+		} else {
+			keys = fmt.Sprintf("%c", key)
+		}
+		return p.send(keys)
+	case "On":
+		key := fmt.Sprintf("%c", key)
+		p.normalBuffer = addKeyToBuffer(p.normalBuffer, key)
+	case "Off":
+		key := fmt.Sprintf("%c", key)
+		p.normalBuffer = removeKeyFromBuffer(p.normalBuffer, key)
 	}
-
-	var keys string
-	if unicode.IsUpper(key) {
-		keys = fmt.Sprintf("shift-%c", unicode.ToLower(key))
-	} else {
-		keys = fmt.Sprintf("%c", key)
-	}
-
-	return p.send(keys)
+	return nil
 }
 
 func (p *proxmoxDriver) SendSpecial(special string, action bootcommand.KeyAction) error {
 	keys := special
 	if replacement, ok := p.specialMap[special]; ok {
 		keys = replacement
 	}
-
-	return p.send(keys)
+	switch action.String() {
+	case "Press":
+		return p.send(keys)
+	case "On":
+		p.normalBuffer = addKeyToBuffer(p.normalBuffer, keys)
+	case "Off":
+		p.normalBuffer = removeKeyFromBuffer(p.normalBuffer, keys)
+	}
+	return nil
 }
 
-func (p *proxmoxDriver) send(keys string) error {
-	err := p.client.Sendkey(p.vmRef, keys)
+func (p *proxmoxDriver) send(key string) error {
+	keys := append(p.specialBuffer, p.normalBuffer...)
```

Comment:
Just to check - is there a need for the dual buffers, since it apparently worked without?

Reply:
yeah, there is. But I don't think my test case covered that. The thing is that we need to send modifiers first. otherwise we could end up sending something like `n-shift` instead of `shift-n`. Proxmox will then receive `n` instead of `N`. That's why I separated the buffers.

Reply:
This problem appears when holding down an alphabetical key then shift and after that send another letter.
Something like this:
```
<cOn><leftShiftOn>n<leftShiftOff><cOff>
```
If we had only one buffer the key event `c-shift-n` will be sent to Proxmox and result in a "cN"

Reply:
Ah, okay! Then nice catch, and good with the added test case!